### ISSUE TITLE
[Question] Processing single quotes after SmartyPants

### ISSUE THREAD

**Issue by cincuranet** [lunet-io/markdig#125]
I'm trying to convert all single quotes left after _SmartyPants_ extension into `’` (`&rsquo;`), but I'm failing. Whenever I introduce some `LiteralInline` into `processor` while in `Match` method, I erase everything what _SmartyPants_ did. I think I need bit help where to properly plug the extension and how to match the char. Because so far all my attempts with my `InlineParser` failed.

This code:
```
Markdown.ToHtml("\"hello' test\"", pipeline);
```
returns `<p>&ldquo;hello' test&rdquo;</p>` (with _SmartyPants_). So far so good. What I'd like to archieve is to get `<p>&ldquo;hello&rsquo; test&rdquo;</p>` (or `<p>&ldquo;hello’ test&rdquo;</p>`).

Thanks for any kick in right direction.

**Comment by xoofx** [comment]
Not sure that would be easy to do it outside the existing extensions that already process these characters. I would start to modify the code at [`SmartyPantsInlineParser:L282`](https://github.com/lunet-io/markdig/blob/9906a0554fa0d07a21e6d1fedb0c04743ea58596/src/Markdig/Extensions/SmartyPants/SmartyPantsInlineParser.cs#L282) to keep the characters you want based on the criteria you are expecting...

**Comment by cincuranet** [comment]
Hmm. _SmartyPants_ is using `OpeningCharacters` while my `'` can be anywhere (see my example). Your idea of directly changing the parser is viable. I can disable then the balancing checking.

**Comment by cincuranet** [comment]
No, the balancing works only if already in between quotes. Here the _SmartyPants_ does not fire and hence my change does not apply. :(

```
Markdown.ToHtml("hello' test", pipeline);
```

**Comment by xoofx** [comment]
The `OpeningCharacters` is just used to indicate that the parser must be called for this character, but it can happen anywhere, that's fine. You just have to change slightly the behavior in the method `BlockOnProcessInlinesEnd` to better handle your case.

**Comment by cincuranet** [comment]
Finally somewhat hacked it together. There's not much changes at the end. Here's the diff (some whitespaces are fucked up, I know): https://github.com/cincuranet/markdig/commit/192b875055f349b58057fa8f1582581d97cedfb5?w=1 .

Not sure whether it's too far from original _SmartyPants_ idea or not and hence some PR would be worth.
